pkg/controller/pcidevice/pcidevice_controller.go

```diff
@@ -52,16 +52,17 @@ func (h Handler) reconcilePCIDevices(hostname string) error {
 	var setOfRealPCIAddrs map[string]bool = make(map[string]bool)
 	for _, dev := range pci.Devices {
 		setOfRealPCIAddrs[dev.Address] = true
-		name := v1beta1.PCIDeviceNameForHostname(dev, hostname)
+		name := v1beta1.PCIDeviceNameForHostname(dev, nodename)
 		// Check if device is stored
 		_, err := h.client.Get(name, metav1.GetOptions{})
 
 		if err != nil {
```

Comment:
Not introduced in this PR, but if we want to check if a resource exists or not, we can check the error with `apierrors.IsNotFound` or `apierrors.IsAlreadyExists`, and return the error if they are other errors. For example, https://github.com/harvester/harvester/blob/f47ad39308607f92b3b8b8758b87258bd8ac3ac2/pkg/controller/master/backup/backup_target.go#L165-L167

Reply:
I introduced it as a comment TODO